app/controllers/posts_controller.rb

```diff
@@ -19,9 +22,24 @@ def clap
     end
   end
 
+  def preview
+    render :preview
+  end
+
   private
 
   def set_post
     @post = Post.published.friendly.find params[:id]
   end
+
+  def set_post_preview
+    @post = Post.new(post_params)
+    @post.user = current_admin_user
+    @post.created_at = Time.now
+    @post.id = 99999
```

Comment:
INFO: Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them.